chain/near/src/chain.rs

```diff
@@ -0,0 +1,383 @@
+        _loc: &DeploymentLocator,
+        _capabilities: &Self::NodeCapabilities,
+        _unified_api_version: UnifiedMappingApiVersion,
+        _stopwatch_metrics: StopwatchMetrics,
+    ) -> Result<Arc<Self::TriggersAdapter>, Error> {
+        let adapter = TriggersAdapter {};
+        Ok(Arc::new(adapter))
+    }
+
+    async fn new_block_stream(
+        &self,
+        deployment: DeploymentLocator,
+        start_blocks: Vec<BlockNumber>,
+        filter: Arc<TriggerFilter>,
+        metrics: Arc<BlockStreamMetrics>,
+        unified_api_version: UnifiedMappingApiVersion,
+    ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        if start_blocks.len() != 0 && start_blocks.len() != 1 {
+            return Err(anyhow::format_err!(
+                "accepting start_blocks length of 0 or 1, got {}",
+                start_blocks.len()
+            ));
+        }
+
+        let adapter = self
+            .triggers_adapter(
+                &deployment,
+                &NodeCapabilities {},
+                unified_api_version.clone(),
+                metrics.stopwatch.clone(),
+            )
+            .expect(&format!("no adapter for network {}", self.name,));
+
+        let firehose_endpoint = match self.firehose_endpoints.random() {
+            Some(e) => e.clone(),
+            None => return Err(anyhow::format_err!("no firehose endpoint available",)),
+        };
+
+        let logger = self
+            .logger_factory
+            .subgraph_logger(&deployment)
+            .new(o!("component" => "FirehoseBlockStream"));
+
+        let firehose_mapper = Arc::new(FirehoseMapper {});
+        let firehose_cursor = self.subgraph_store.writable(&deployment)?.block_cursor()?;
+
+        Ok(Box::new(FirehoseBlockStream::new(
+            firehose_endpoint,
+            firehose_cursor,
+            firehose_mapper,
+            // FIXME (NEAR): Hard-coded NodeId, this is actually not required for other chain ...
+            NodeId::new("near").unwrap(),
+            deployment.hash,
+            adapter,
+            filter,
+            start_blocks,
+            logger,
+        )))
+    }
+
+    fn ingestor_adapter(&self) -> Arc<Self::IngestorAdapter> {
+        let logger = self
+            .logger_factory
+            .component_logger(
+                "BlockIngestor",
+                Some(ComponentLoggerConfig {
+                    elastic: Some(ElasticComponentLoggerConfig {
+                        index: String::from("block-ingestor-logs"),
+                    }),
+                }),
+            )
+            .new(o!());
+
+        let adapter = IngestorAdapter { logger };
+        Arc::new(adapter)
+    }
+
+    fn chain_store(&self) -> Arc<dyn ChainStore> {
+        self.chain_store.clone()
+    }
+
+    async fn block_pointer_from_number(
+        &self,
+        _logger: &Logger,
+        _number: BlockNumber,
+    ) -> Result<BlockPtr, IngestorError> {
+        // FIXME (NEAR): Hmmm, what to do with this?
```

Comment:
Isn't it also required in the querying layer? And not just for start block?

I think it was from what I recall David told me.

Reply:
For queries it goes through `QueryStore::block_number` which is a different code path. It might be good to check what that is doing for Near, maybe it just assumes it's dealing with an Ethereum block number. This is important but can left to a future PR.